### DISCUSSION TITLE
How to disable fading effect?

### DISCUSSION THREAD

**Discussion by MunifTanjim** [nvim-neo-tree/neo-tree.nvim#267] [Q&A]
Sorry if it's already documented somewhere.

I tried to look into https://github.com/nvim-neo-tree/neo-tree.nvim/pull/227 but couldn't find/understand how to disable the fading effect.

Is there any way I can do it?

Asking because sometimes I navigate the file tree like a normal buffer and press `$` to go to the end of line to see the whole filename.

**Comment by gegoune** [comment]
@MunifTanjim Might be jumping a gun here, but try `resize_timer_interval = -1`.

**Reply by cseickel** [reply]
Close. that will prevent it from updating on resize but it won't skip the initial fading.

**Comment by cseickel** [comment]
You'll need to override the renderer with your own layout that does not include any containers.
https://github.com/nvim-neo-tree/neo-tree.nvim/blob/d8aa8ea5edcbddc2d97e4582f63b2205c5733c7f/lua/neo-tree/defaults.lua#L114-L163


Here is how to have the default layout with the container parts deleted:
```
require("neo-tree").setup({
  renderers = {
    directory = {
      { "indent" },
      { "icon" },
      { "current_filter" },
      { "name" },
      { "clipboard" },
      { "diagnostics", errors_only = true },
    },
    file = {
      { "indent" },
      { "icon" },
      {
        "name",
        use_git_status_colors = true,
        zindex = 10
      },
      { "clipboard" },
      { "bufnr" },
      { "modified" },
      { "diagnostics" },
      { "git_status" },
    },
  },
})
```

This does mean that you'll lose the right aligned symbols.

**Reply by MunifTanjim** [reply]
This is not working for some reason 🤔 

This is the config I'm using: https://github.com/MunifTanjim/dotfiles/blob/54eef761b9e5650bd69d004ed54280cb5da9e6b0/private_dot_config/nvim/lua/config/neo-tree.lua

**Reply by cseickel** [reply]
It's just a nesting error. `file` should be part of the `renderers` table above it, along with `directory`:
https://github.com/MunifTanjim/dotfiles/blob/313390c811fdf647bf5bfdc3752a3a68d48e6839/private_dot_config/nvim/lua/config/neo-tree.lua#L156

**Reply by MunifTanjim** [reply]
Oh, I fixed that and updated the link above. Still getting the same result.

**Reply by cseickel** [reply]
There seems to be a bug where setting renderers at the global level is not working. If you move that key up into the `filesystem` config it will work though.

I created an issue for this bug: https://github.com/nvim-neo-tree/neo-tree.nvim/issues/291

**Reply by nyngwang** [reply]
@cseickel Could you also explain the purpose of `current_filter` here?

**Reply by cseickel** [reply]
@nyngwang When you are filtering the tree, that will display the search term being used. Otherwise it does nothing.

**Reply by nyngwang** [reply]
@cseickel It seems that a missing `{` is needed below the `{ "current_filter" }` on the code above?

**Reply by cseickel** [reply]
@nyngwang They actually had the wrong indentation and an extra dangling `}` at teh end of each renderer from where I deleted the container components. I fixed the code snippets.
